prow/config/jobs/sail-operator.yaml

```diff
@@ -0,0 +1,68 @@
+org: istio-ecosystem
+repo: sail-operator
+branches:
+  - pre-main
+  - main
+
+support_release_branching: false
+image: gcr.io/istio-testing/build-tools:master-ded7e39747b2cf9eb9fa8c2dbcf25d9c84a4b407
+jobs:
+  - name: unit-tests
+    types: [presubmit]
+    architectures: [amd64, arm64]
+    command: [entrypoint, make, -e, "T=-v -count=1", build, test]
+
+  - name: gencheck
+    types: [presubmit]
+    command: [make, gencheck]
+
+  - name: lint
+    types: [presubmit]
+    command: [make, lint]
+    resources: lint
+
+  - name: integ-kind
+    types: [presubmit]
+    command: [entrypoint, make, test.integration.kind]
+    requirements: [kind]
+
+  - name: scorecard
+    types: [presubmit]
+    command: [entrypoint, make, test.scorecard]
+    requirements: [kind]
+
+  - name: update-deps
+    types: [periodic]
+    cron: "0 23 * * 1-5" # run Mon-Fri at 11:00PM UTC
+    command:
+    - entrypoint
+    - ../test-infra/tools/automator/automator.sh
+    - --org=$AUTOMATOR_ORG
+    - --repo=sail-operator
+    - "--title=Automator: Update dependencies in $AUTOMATOR_ORG/$AUTOMATOR_REPO@$AUTOMATOR_BRANCH"
+    - --labels=auto-merge
+    - --modifier=update_deps
+    - --token-env
+    - --cmd=./tools/update_deps.sh
+    requirements: [github-istio-testing]
```

Comment:
This gives (very) high level privilege here. This is acceptable, as its a periodic job that runs only on approved+merged code, rather than arbitrary presubmit code. We do this in Istio in many places. However, all Istio code like this is protect by CODEOWNERS where only trusted maintainers have approval access.

So I think this is fine as long as we make sure the repo maintains a similar trust level. In aprticular, do we need the 3 third party Admin tools here https://github.com/istio-ecosystem/sail-operator/settings/access? And will a CODEOWNERS be established? When it is, can we ensure that the owners for the `tools/` is scoped only to Istio maintainers?